src/observersTools/leggedOdometryTools.cpp

```diff
@@ -0,0 +1,860 @@
+
+  const so::Matrix3 & realRobotOri = realRobot.posW().rotation().transpose();
+
+  run(ctl, logger, pose, vel, realRobotOri);
+}
+
+void LeggedOdometryManager::run(const mc_control::MCController & ctl, mc_rtc::Logger & logger, sva::PTransformd & pose)
+{
+  const auto & realRobot = ctl.realRobot(robotName_);
+
+  const so::Matrix3 & realRobotOri = realRobot.posW().rotation().transpose();
+  // the tilt must come from another estimator so we use the real robot for the orientation
+  run(ctl, logger, pose, realRobotOri);
+}
+
+void LeggedOdometryManager::run(const mc_control::MCController & ctl,
+                                mc_rtc::Logger & logger,
+                                sva::PTransformd & pose,
+                                sva::MotionVecd & vel,
+                                sva::MotionVecd & acc,
+                                const stateObservation::Matrix3 & tilt)
+{
+  updateJointsConfiguration(ctl);
+  odometryRobot().posW(fbPose_);
+
+  // we set the velocity and acceleration to zero as they will be compensated anyway as we compute the
+  // successive poses in the local frame
+  sva::MotionVecd zeroMotion;
+  zeroMotion.linear() = so::Vector3::Zero();
+  zeroMotion.angular() = so::Vector3::Zero();
+  odometryRobot().velW(zeroMotion);
+  odometryRobot().accW(zeroMotion);
+
+  odometryRobot().forwardKinematics();
+  odometryRobot().forwardVelocity();
+  odometryRobot().forwardAcceleration();
+
+  // detects the contacts currently set with the environment
+  contactsManager().findContacts(ctl, robotName_);
+  // updates the contacts and the resulting floating base kinematics
+  updateFbAndContacts(ctl, logger, true, true, tilt);
+  // updates the floating base kinematics in the observer
+  updateFbKinematics(pose, vel, acc);
+}
+
+void LeggedOdometryManager::run(const mc_control::MCController & ctl,
+                                mc_rtc::Logger & logger,
+                                sva::PTransformd & pose,
+                                sva::MotionVecd & vel,
+                                const stateObservation::Matrix3 & tilt)
+{
+  updateJointsConfiguration(ctl);
+  odometryRobot().posW(fbPose_);
+
+  // we set the velocity and acceleration to zero as they will be compensated anyway as we compute the
+  // successive poses in the local frame
+  sva::MotionVecd zeroMotion;
+  zeroMotion.linear() = so::Vector3::Zero();
+  zeroMotion.angular() = so::Vector3::Zero();
+  odometryRobot().velW(zeroMotion);
+  odometryRobot().accW(zeroMotion);
+
+  odometryRobot().forwardKinematics();
+  odometryRobot().forwardVelocity();
+  odometryRobot().forwardAcceleration();
+
+  // detects the contacts currently set with the environment
+  contactsManager().findContacts(ctl, robotName_);
+  // updates the contacts and the resulting floating base kinematics
+  updateFbAndContacts(ctl, logger, true, false, tilt);
+  // updates the floating base kinematics in the observer
+  updateFbKinematics(pose, vel);
```

Comment:
This is almost the same implementation as the other variant. I would implement this with a private variant of the function that takes a pointer to `vel` and `acc` and adapt the parameters accordingly